packages/esm-patient-chart-app/src/ui-components/forms-side-rail.component.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { Button } from 'carbon-components-react';
+import Document20 from '@carbon/icons-react/es/document/20';
+import styles from './forms-side-rail.scss';
+import { ScreenModeTypes, WindowSize } from '@openmrs/esm-patient-common-lib';
+import Close20 from '@carbon/icons-react/es/close/20';
+import WarningFilled16 from '@carbon/icons-react/es/warning--filled/16';
+import { patientChartWorkspaceSlot } from '../constants';
+import { useAssignedExtensionIds } from '@openmrs/esm-framework';
+
+interface FormsSideRailProps {
+  checkViewMode: (active: boolean) => void;
+  windowSize: WindowSize;
+}
+
+const FormsSideSideRail: React.FC<FormsSideRailProps> = ({ checkViewMode, windowSize }) => {
+  const extensions = useAssignedExtensionIds(patientChartWorkspaceSlot);
+  const [mouseHover, setMouseHover] = useState<boolean>(false);
+
+  const active =
+    !(
+      extensions.filter((ext) => ext === 'order-basket-workspace' || ext === 'visit-notes-form-workspace').length > 0
+    ) && extensions.length > 0;
```

Comment:
Related to my other comment about the workspace API, we might define a function in common-lib, something like

```tsx
function isWorkspaceOpen(workspaceName: string) {
  const extensions = useAssignedExtensionIds(patientChartWorkspaceSlot);
  return extensions.some(ext => ext === workspaceName);
}
```